compiler/rustc_lint/src/useless_send_constraint.rs

```diff
@@ -0,0 +1,56 @@
+use rustc_span::sym;
+
+use crate::hir;
+
+use crate::{lints::UselessSendConstraintDiag, LateContext, LateLintPass};
+
+declare_lint! {
+    /// The `lint_useless_send_constraint` lints useless constraint of references to `Send`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// fn foo(_: &(dyn Any + Send>) {}
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// References cannot be sent across threads unless they have a `Sync` bound, so constraining them to `Send` without `Sync` is useless.
+    pub USELESS_SEND_CONSTRAINT,
```

Comment:
nit wrt naming:

I prefer "unnecessary" over "useless" here. useless feels a lot more negative to me and even sounds like it kind of blames the author for writing bad code :thinking:

Reply:
done

Reply:
please also update the PR name to reflect this change

Reply:
done